### PR TITLE
Release 0.41.0

### PR DESCRIPTION
**Pull request by zippolyte**
Also add release action and adjust RELEASING.md

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+on:
+  pull_request:
+  release:
+    types:
+      - published
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
```
Comment (nmuesch) [inline .github/workflows/release.yaml:11]: Where is this matrix?

Comment (zippolyte) [reply .github/workflows/release.yaml:11]: I'm returning you the question hahaha

Comment (zippolyte) [reply .github/workflows/release.yaml:11]: https://github.com/DataDog/datadog-api-client-python/blob/cdd884303b2c2d7959cff6890a7a36d1173bd0b6/.github/workflows/release.yml#L11

**Review by nmuesch — APPROVED** [review]

**Pull request merged**